io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -404,14 +404,13 @@ impl Lvol {
 
         Ok(())
     }
-
     async fn do_create_snapshot(
         &self,
         snap_param: SnapshotParams,
         done_cb: unsafe extern "C" fn(*mut c_void, *mut spdk_lvol, i32),
         done_cb_arg: *mut ::std::os::raw::c_void,
-        receiver: Option<oneshot::Receiver<i32>>,
-    ) -> Result<(), Error> {
+        receiver: Option<oneshot::Receiver<(i32, Lvol)>>,
```

Comment:
why is receiver an option?

Reply:
> why is receiver an option?

it is used in create_snapshot_remote() function where no receiver parameter is used. create_snapshot() need receiver parameter as We need to get the lvol as part of SPDK callback. I guess similar things to be done in remote snapshot create (As part of NexusCreateSnapshot), which is not part of current PR scope.

Reply:
Ah, I see... in such case I suggest also having 2 functions, one with a receiver and another without a receiver at all, and the one with the receiver call the function without the receiver, if you see what I mean?

Reply:
ok.. let me check if create_snapshot_remote() also need a receiver to get the snapshot UUID in callback. In that case, We can use single function. Else will use different function for local and remote snapshot create.